### ISSUE TITLE
Progress of overall extraction.

### ISSUE THREAD

**Issue by LiamKarlMitchell** [maxogden/extract-zip#8]
I am using this module in a game launcher/patcher.

I would like to implement a way to get the current file being extracted and overall progress of an extraction operation, from what I can tell the zip parser is quite nice.

Im just thinking of doing a fork and adding this as an extra option.
Thought I would post here in-case there are any guidelines or gotchas someone might recommend.

Would use an event emitter to notify about the current status and progress.
Time permitting I will do this soon.

Thanks for the wrapper :).

**Comment by maxogden** [comment]
Rather than an event emitter I would do it as a static property that can be polled on an interval, e.g. have this function return an object https://github.com/maxogden/extract-zip/blob/master/index.js#L9 so you could do:

```
var extract = extractZip(myZip, cb)
setInterval(log, 250)

function log () {
  console.log(extract.finished / extract.total + '%')
}
```

it's a little weird to have to return something *and* have a callback in the same API, but the only other option would be to pass an object in that gets the progress properties set on it, and I feel like that would be even weirder.

**Comment by LiamKarlMitchell** [comment]
Ah yeah, that is probably less over head thanks :)

**Comment by McFarts** [comment]
@maxogden  thanks for that, but extract is returning undefined for me, any idea? :(

**Comment by malept** [comment]
Fixed by #8.
